File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C/s527301188_comments_removed.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char input[1024];
    if (fgets(input, sizeof(input), stdin) == NULL) {
        return 1;
    }

    char *strs[100];
    int nums[100];
    int top = 0;

    strs[0] = strtok(input, " \n");
    while (strs[top] != NULL) {
        switch (strs[top][0]) {
            case '+':
                top--;
                nums[top-1] = nums[top - 1] + nums[top];
                break;
            case '-':
                top--;
                nums[top-1] = nums[top - 1] - nums[top];
                break;
            case '*':
                top--;
                nums[top-1] = nums[top - 1] * nums[top];
                break;
            default:
                nums[top] = atoi(strs[top]);
                top++;
                break;
        }
        strs[top] = strtok(NULL, " \n");
    }

    printf("%d\n", nums[0]);

    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Go/s527301188_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("Error reading input:", err)
		return
	}

	strs := strings.Fields(input)
	nums := make([]int, len(strs))
	top := 0

	for i := 0; i < len(strs); i++ {
		switch strs[i][0] {
		case '+':
			if top < 2 {
				fmt.Println("Invalid operation: not enough operands for +")
				return
			}
			top--
			nums[top-1] = nums[top-1] + nums[top]
		case '-':
			if top < 2 {
				fmt.Println("Invalid operation: not enough operands for -")
				return
			}
			top--
			nums[top-1] = nums[top-1] - nums[top]
		case '*':
			if top < 2 {
				fmt.Println("Invalid operation: not enough operands for *")
				return
			}
			top--
			nums[top-1] = nums[top-1] * nums[top]
		default:
			num, err := strconv.Atoi(strs[i])
			if err != nil {
				fmt.Println("Error converting string to int:", err)
				return
			}
			nums[top] = num
			top++
		}
	}

	if top != 1 {
		fmt.Println("Invalid input: too many or too few operands")
		return
	}

	fmt.Println(nums[0])
}

// End of Code
